### PR TITLE
WT-9143 Refactor and review existing CppSuite bounded cursor stress for improvement

### PR DESCRIPTION
**Pull request by clarissecheah**
- Renamed existing bounded cursor stress tests
- Refactored bounded_cursor_stress to use bound_set
- Added an apply() method in the bound class

### CODE REVIEW

File: dist/test_data.py
Comment (jiechenbo) [inline dist/test_data.py:249]: np: alphabetize

File: test/cppsuite/tests/bounded_cursor_stress.cpp
Comment (jiechenbo) [inline test/cppsuite/tests/bounded_cursor_stress.cpp:649]: I would just use bound_set instead of separating to lower bound and upper bound.

File: test/suite/test_cursor_bound_fuzz.py
Comment (jiechenbo) [inline test/suite/test_cursor_bound_fuzz.py:153]: you can revert the changes done in this file

File: test/evergreen.yml
Comment (jiechenbo) [inline test/evergreen.yml:4269]: np: alphabetize

File: test/cppsuite/tests/bounded_cursor_stress.cpp
Comment (jiechenbo) [inline test/cppsuite/tests/bounded_cursor_stress.cpp:669]: ditto

File: test/cppsuite/tests/bounded_cursor_stress.cpp
Comment (jiechenbo) [inline test/cppsuite/tests/bounded_cursor_stress.cpp:719]: ditto

File: test/cppsuite/tests/bounded_cursor_stress.cpp
Comment (jiechenbo) [inline test/cppsuite/tests/bounded_cursor_stress.cpp:224]: So i just realised that we can place a ternary `? :` operator here for reverse collator, could you try that out here?

File: test/cppsuite/tests/bounded_cursor_stress.cpp
Comment (jiechenbo) [inline test/cppsuite/tests/bounded_cursor_stress.cpp:218]: realised we don't need the braces here anymore

File: test/cppsuite/tests/bounded_cursor_stress.cpp
Comment (jiechenbo) [inline test/cppsuite/tests/bounded_cursor_stress.cpp:214]: don't need the braces here anymore

**Review by jiechenbo — COMMENTED** [review]
Nice job!! Have a few comments, generally mostly alphabetizing

File: test/cppsuite/tests/bounded_cursor_stress.cpp
Comment (clarissecheah) [reply test/cppsuite/tests/bounded_cursor_stress.cpp:649]: Addressed this and other review comments here:
[c0b60a7](https://github.com/wiredtiger/wiredtiger/pull/8223/commits/c0b60a7e679a4b383f5507dd0caf13a8467c4c37)

**Review by luke-pearson — APPROVED** [review]
LGTM, big improvement, thanks.

File: dist/test_data.py
Comment (etienneptl) [inline dist/test_data.py:249]: np: Can we alphabetize this list?

File: test/cppsuite/tests/bounded_cursor_stress.cpp
Comment (etienneptl) [inline test/cppsuite/tests/bounded_cursor_stress.cpp:203]: Extra space

File: test/cppsuite/tests/bounded_cursor_stress.cpp
Comment (etienneptl) [inline test/cppsuite/tests/bounded_cursor_stress.cpp:235]: We can avoid the extra previous line:
```suggestion
        return bound_set(lower_bound, upper_bound);
```

File: test/cppsuite/tests/bounded_cursor_stress.cpp
Comment (etienneptl) [inline test/cppsuite/tests/bounded_cursor_stress.cpp:247]: I think we can pass by reference `bounds` (same for `validate_bound_search_near`):
```suggestion
      const std::string &search_key, const bound_set &bounds)
```

File: test/cppsuite/tests/run.cpp
Comment (etienneptl) [inline test/cppsuite/tests/run.cpp:139]: np: alphabetize

File: test/cppsuite/tests/run.cpp
Comment (etienneptl) [inline test/cppsuite/tests/run.cpp:182]: np: alphabetize (sorry...)

**Review by etienneptl — APPROVED** [review]
Sorry for the late review, LGTM after a few nitpicks!

File: dist/test_data.py
Comment (clarissecheah) [reply dist/test_data.py:249]: I think this is alphabetized, but I will fix the rest you mentioned. :)

Comment (clarissecheah) [reply dist/test_data.py:249]: [cbad707](https://github.com/wiredtiger/wiredtiger/pull/8223/commits/cbad70778d40291639468372f21fe74e194e0e09)

File: test/cppsuite/tests/bounded_cursor_stress.cpp
Comment (jiechenbo) [inline test/cppsuite/tests/bounded_cursor_stress.cpp:223]: np: we can remove braces here

File: test/cppsuite/tests/bounded_cursor_stress.cpp
Comment (jiechenbo) [inline test/cppsuite/tests/bounded_cursor_stress.cpp:230]: np: i would remove the braces around `lower_bound.get_key()`, since its not needed

**Review by jiechenbo — APPROVED** [review]
LGTM! have a few nitpicks but overall looks good. Please run a patch build before merging this change to make sure that all is good 👍

File: test/cppsuite/tests/bounded_cursor_stress.cpp
Comment (clarissecheah) [reply test/cppsuite/tests/bounded_cursor_stress.cpp]: [0270585](https://github.com/wiredtiger/wiredtiger/pull/8223/commits/02705857162292b83da6f37561cdbcce7636cbb8)

**Pull request merged**
